calendar-app/src/api/saveAndEditEvent.ts

```diff
@@ -0,0 +1,87 @@
+import { StoredEvent, RequestResult } from "../utils/types";
+import { SERVER_URL } from "../pages/calendar/MainContent/consts";
+import apiRequest from "./sendAPIRequest";
+import { addEvent, updateEvent } from "../features/eventDisplay";
+import { store } from "../store";
+
+async function saveEvent(
+  form: StoredEvent,
+  id?: number
+): Promise<RequestResult> {
+  const isTitleEntered = !form.title || form.title.trim() === "";
+
+  const areDatesEntered = !form.startDate || !form.startTime || !form.endTime;
+
+  const isEndAfterStart =
+    form.startDate > form.endDate ||
+    !(form.startDate === form.endDate && form.startTime > form.endTime);
+
+  if (isTitleEntered) {
+    alert("Title is mandatory.");
+    return { success: false, error: "Title is mandatory." };
+  } else if (areDatesEntered) {
+    alert("Make sure to enter start and end time and date.");
+    return {
+      success: false,
+      error: "Make sure to enter start and end time and date.",
+    };
+  } else if (!isEndAfterStart) {
+    alert("End time of event can not be before the start time.");
+    return {
+      success: false,
+      error: "End time of event can not be before the start time.",
+    };
+  }
+  if (form.endDate.toString() === "Invalid Date" || form.endDate === "") {
+    form.endDate = form.startDate;
+  }
+  form.eventKey = `${new Date(form.startDate).getDay()}_${form.startTime.slice(
```

Comment:
mutation of `form` object passed via arguments, this can have unexpected side-effects